base/src/commonMain/kotlin/monitor/ServiceMonitor.kt

```diff
@@ -17,36 +17,54 @@
 
 package com.splendo.kaluga.base
 
+import com.splendo.kaluga.base.monitor.ServiceMonitorState
 import com.splendo.kaluga.logging.debug
-import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.asStateFlow
+import com.splendo.kaluga.state.ColdStateFlowRepo
+import kotlin.coroutines.CoroutineContext
 
 interface ServiceMonitor {
-    val isServiceEnabled: Boolean
-    val isEnabled: Flow<Boolean>
     fun startMonitoring()
     fun stopMonitoring()
 }
 
-abstract class DefaultServiceMonitor : ServiceMonitor {
+abstract class DefaultServiceMonitor(
+    coroutineContext: CoroutineContext
+) : ColdStateFlowRepo<ServiceMonitorState>(
+    coroutineContext = coroutineContext,
+    initChangeStateWithRepo = { state, repo ->
+        debug("DefaultServiceMonitor") { "initChangeStateWithRepo with $state" }
+        (repo as ServiceMonitor).run {
```

Comment:
https://github.com/splendo/kaluga/pull/385/files#r866812863 this will need to be `repo as? DefaultServiceMonitor` to resolve that comment.. but that is what it pretty much has to be always.